PNPM: MissingFieldException: Field 'version' is required

### Describe the bug

Analyzing a project using the Analyzer PNPM, with ORT 38.0.0 I get the following stacktrace for a repository that previously could be scanned without issues:

```
kotlinx.serialization.MissingFieldException: Field 'version' is required for type with serial name 'org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo', but it was missing at path: $[1]
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:95)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableValue(AbstractDecoder.kt:43)
	at kotlinx.serialization.encoding.AbstractDecoder.decodeSerializableElement(AbstractDecoder.kt:70)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:168)
	at kotlinx.serialization.encoding.CompositeDecoder$DefaultImpls.decodeSerializableElement$default(Decoding.kt:538)
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(CollectionSerializers.kt:80)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.readElement$default(CollectionSerializers.kt:51)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:36)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:43)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:69)
	at kotlinx.serialization.json.Json.decodeFromString(Json.kt:165)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfoKt.parsePnpmList(ModuleInfo.kt:50)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.Pnpm.getWorkspaceModuleDirs(Pnpm.kt:110)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.Pnpm.resolveDependencies(Pnpm.kt:77)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.Pnpm.resolveDependencies(Pnpm.kt:70)
	at org.ossreviewtoolkit.analyzer.PackageManager.resolveDependencies(PackageManager.kt:295)
	at org.ossreviewtoolkit.analyzer.PackageManagerRunner$run$3.invokeSuspend(Analyzer.kt:321)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:101)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:113)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
{
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:589)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:823)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:720)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:707)
Caused by: kotlinx.serialization.MissingFieldException: Field 'version' is required for type with serial name 'org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo', but it was missing
	at kotlinx.serialization.internal.PluginExceptionsKt.throwMissingFieldException(PluginExceptions.kt:20)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo.<init>(ModuleInfo.kt:29)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo$$serializer.deserialize(ModuleInfo.kt:29)
	at org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo$$serializer.deserialize(ModuleInfo.kt:29)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue(StreamingJsonDecoder.kt:69)
```

In the web-app-report, I get an issue as follows:

```
PNPM failed to resolve dependencies for path 'ui/xxx/package.json': MissingFieldException: Field 'version' is required for type with serial name 'org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo', but it was missing at path: $[1]
Caused by: MissingFieldException: Field 'version' is required for type with serial name 'org.ossreviewtoolkit.plugins.packagemanagers.node.pnpm.ModuleInfo', but it was missing
```
A first analysis showed that actually unter this path there are `package.json` files that don't have a `version` field.

### Expected behavior

I would expect that existing repository scans still can be scanned after refactorings without issues.
But there might be reasons why this version field is mandatory now. If yes, please document in this ticket.

### Temporary workaround

Make sure all `package.json` files in the repository to be scanned have a `version` field.

### Environment

* ORT version: 38.0.0

--- a/plugins/package-managers/node/src/main/kotlin/pnpm/ModuleInfo.kt
+++ b/plugins/package-managers/node/src/main/kotlin/pnpm/ModuleInfo.kt
@@ -28,8 +28,8 @@ internal fun parsePnpmList(json: String): List<ModuleInfo> = JSON.decodeFromStri
 
 @Serializable
 internal data class ModuleInfo(
-    val name: String,
-    val version: String,
+    val name: String? = null,
+    val version: String? = null,
     val path: String,
     val private: Boolean,
     val dependencies: Map<String, Dependency> = emptyMap(),

